golang/inbound.go

```diff
@@ -166,6 +166,15 @@ func (c *Connection) dispatchInbound(call *InboundCall) {
 		return
 	}
 
+	// TODO(prashant): This is an expensive way to check for cancellation, and is not thread-safe.
```

Comment:
I was thinking at the time that maybe we'd maintain a heap of exchanges ordered by deadline, and have a ticker thread that just checked the heap and purged any exchanges that had expired.